### ISSUE TITLE
Support for images with transparency

### ISSUE THREAD

**Issue by Derp0** [Zarh/ManaGunZ#68]
Could you please add support for transparency?
Currently if I put a PNG with transparency in game ISO folders Managunz stretches them (or in the case of PS2ISO it refuses to show any covers/icons).

**Comment by Zarh** [comment]
Hi, thank you for the feedback.
It should support the transparency, I need more information.
Can you send me the picture you want to display ?
Which picture are you talking about, the main picture in XMB UI ? Maybe you are using another UI ?)
Does it have the same filename as the game iso ? 

> For example : 
> Backup path : "/dev_hdd0/PS2ISO/MyGame.iso"
> Picture path : "/dev_hdd0/PS2ISO/MyGame.png"

**Comment by Syrus-33** [comment]
Make sure the size of the png is 320x176 and has the exact naming of the iso as stated above.

**Comment by Derp0** [comment]
OK, let's explain them one by one because it's a different problem for every game type.

# 1. PS3 games

https://i.imgur.com/beZRuoY.png

The cover on the left is the same as ICON0, but with added transparency (i.e. 320x320). It's not terrible, but both images are stretched, that is they are not displayed as 320x176. I used save screenshot function from XMB and opened the resulting screenshot in Photoshop to confirm. I have no idea whether this method is 100% accurate or not.

# 1. PS2 games

https://i.imgur.com/1jXXagN.png

I put the cover on the left inside Managunz "cover" directory and used "Create ICON0" function from the Game Menu to create the icon. The problem is if I take that icon over to Photoshop and add transparency and replace the the one created by Managunz itself then nothing is displayed anymore (neither cover on the left nor the icon).

Why do I add transparency then you might ask. Well, that is to stop Webman mod from stretching them. This way they are shown correctly in Webman mod. I also posted this over at psx place recently in webMAN MOD - General Information Thread. I've also mentioned you there.

# 1. PSP games

https://i.imgur.com/2TmevHU.png

The cover on the left is 320x320 (With transparency) in size, but as you can see it is horribly overstretched.


P.S. Covers used are in PNG format. If you want them I can upload them as well.

**Comment by Zarh** [comment]
I think I  understand now. 
Can you upload the 3 covers too, please.

I think I'll add an option to not stretch the picture.

**Comment by Zarh** [comment]
Ok, it's done for the next update.
Picture with the size 320x320 won't be stretched. Also, they won't be able to be displayed with game case even if the option is enabled.

**Comment by Derp0** [comment]
Thanks.
You mean if it's anything other than 320x320 it'll be stretched (like 250x250)? If so, I think it'd be better to extend that option to square images because not every one uses that specific size.

> Also, they won't be displayed with game case even if the option is enabled.

Does that only apply to PSX and PS2 games? I'd prefer to have both covers and icons (ICON0 from game ISO) for PSP and PS3 games.
Could you please make it so it'll show the image as cover if I put it in Managunz cover folder and as icon if it's put alongside the ISO?

Also I forgot to ask, does Managunz support image sizes bigger than 1080p for viewing (Not as cover)? I just opened a 4k image and it froze.

**Comment by Zarh** [comment]
Yes, it's a real thing, I didn't found the patch, it come from this : https://blog.madnation.net/ps3-bt-usb-audio-passthrough/
ManaGunZ get the plugin, patch it and store inside his directory. When you enable the system will used the patched libaudio patched instead of the original (with a redirection of path).

Yes it will be display as cover.

There is a memory issue with the picture viewer, I didn't allocate enough memory... it's need to be adjusted.  For example, a 4k picture. it's : 3840 x 2160 x 4 ~ 34MB ... I think I reserved 16 MB for the picture viewer.
there is barely 233MB (supposed to be 256Mo)

I re-opened the issue, I thought 320x320 was generic, I'll apply this rules to every square images exept for PS1.

**Comment by Derp0** [comment]
Just tried with a DS4, but no audio :(
But I'm on 4.86, maybe that's why.

> There is a memory issue with the picture viewer, I didn't allocate enough memory

No worries, I can still view them via XMB.

> I'll apply this rules to every square images exept for PS1.

What's the reason for this exception?

BTW, when I first installed Managunz I had no idea covers had to be transferred to its covers folder, and not game ISO folder for "create ICON0" option to work. I put cover alongside the ISO and kept pressing on that option thinking it was broken when nothing happened. Specifying that in the help section would be nice.

**Comment by Zarh** [comment]
PS1 front cover are a square too.

About the audio, i'll try again maybe it's an issue introduced in latest update....

About create ICON0, I never though about this scenario that's why it's not working.  I always used 'download covers' option, so my covers are inside ManaGunZ directory. Anyway, I opened an issue : https://github.com/Zarh/ManaGunZ/issues/70

**Comment by Derp0** [comment]
> About the audio, i'll try again maybe it's an issue introduced in latest update

What firmware version did you test it on before? What is the code on the back of your DS4?

I don't connect my PS3 to the internet so I have to transfer things manually. I even put IRD files inside GAMES folder myself so that Managunz can use them (Thank you so much for this ability btw, saves me a lot of time).

**Comment by Zarh** [comment]
ok, I added the support of any square picture. they won't be stretched but scaled ;)

(It might be stretched a little bit, but this is caused by something else : the virtual resolution of tiny3D (512x800). To change the size of the virtual resolution, or to simply remove it, I need to reposition every item I added during the past years... When I started the project I didn't know the esthetical issue caused by this... 
It's lot of work... I hope someday I'll have the courage to do it.)

**Comment by Derp0** [comment]
OK, I just realised you've added this feature to 1.41 test build from the other issue. Been testing it for a few hours. I'm very happy with how things look now, awesome work.

Just a question:
Why is it not able to retain a 1:1 ratio? Instead of scaling 320x320 to 294x294 it scales it to 294x274. Is this due to the limitation you mentioned earlier?

**Comment by Zarh** [comment]
Yes, exactly.

The whole project stretched to the resolution of your screen (most of the case it's 1920x1080) from 800x512 the virtual resolution. So, If I display a square in the virtual resolution. It won't be a square anymore in the screen.

Why did I use the virtual resolution, because when I started the project, I didn't know what it was. In tiny3D samples, it is advised to use the virtual resolution to have something nice on every TV, 4:3 or 16:9. It seems to be a nice idea... but now I realized it wasn't, especially for ManaGunZ.

**Comment by Zarh** [comment]
@Derp0 
I just tested "audio through gamepad", it's working. The patched library use voice chat instead of TV output.

I tested it with DS4 v2 with a headset plugged on it, I plugged the gamepad (with a wire).
Then in XMB go to setting>Audio Device Settings and select Wireless controller. Then mount the game with BT audio output with managunz and launch it .

It's probably also working with bluetooth headset, but you must register and select the device in Audio Device Settings
